smartapps/bangali/rooms-child-app.src/rooms-child-app.groovy

```diff
@@ -2410,27 +2413,165 @@ def processCoolHeat()       {
 
 def processCoolHeat()       {
     ifDebug("processCoolHeat")
-    def temp = -1
-    def child = getChildDevice(getRoom())
+       
+    //Need to stop things even if nobody's in
     def isHere = (personsPresence ? personsPresence.currentValue("presence").contains('present') : false)
     if ((checkPresence && !isHere) || maintainRoomTemp == '4')    {
-        if (checkPresence && !isHere)       {
-            if (['1', '3'].contains(maintainRoomTemp))
-                (useThermostat ? roomThermostat.off() : roomCoolSwitch.off())
-            if (['2', '3'].contains(maintainRoomTemp))
-                (useThermostat ? roomThermostat.off() : roomHeatSwitch.off())
-        }
```

Comment:
turning off cool/heat with non-presence is required - no?

thanks.

Reply:
Yes. It still does that, just had to move it down since we don't just turn the thermostat off and do more processing.

Reply:
looks like it only does that now when it finds a rule? if yes, that wont work.

or am i misreading?

Reply:
lines 2476-2479 take care of it when no rules match

Reply:
i might be missing something but why process the rules if checkpresence is on and the person is not present?

thanks.

Reply:
Nope, no reason. I just wanted to get to the full "off" routine and overlooked that. Will fix, along with making the offset constant.